File: python/백준/부르트 포스/1065 한수.py
n = int(input())

count = 0

for i in range(1, n + 1):  # n을 포함하도록 범위 수정
    s = str(i)
    if len(s) == 1:  # 한 자리 수는 모두 한수
        count += 1
    else:
        # 등차수열 확인
        dif = int(s[0]) - int(s[1])
        flag = True
        for j in range(1, len(s)):
            if int(s[j - 1]) - int(s[j]) != dif:
                flag = False
                break
        if flag:
            count += 1

print(count)

File: cpp/백준/부르트 포스/1065 한수.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    int count = 0;

    for (int i = 1; i <= n; ++i) {
        string s = to_string(i);
        if (s.size() == 1) {
            count++;
        } else {
            int dif = (s[0] - '0') - (s[1] - '0');
            bool flag = true;
            for (int j = 1; j < (int)s.size(); ++j) {
                if ((s[j - 1] - '0') - (s[j] - '0') != dif) {
                    flag = false;
                    break;
                }
            }
            if (flag) count++;
        }
    }

    cout << count << "\n";
    return 0;
}
